### PR TITLE
Make lazylists immutable

### LINKED ISSUE #833: Make lists immutable
Jyxal already does this and it'd make implementing copying and stuff way easier. This *would* break answers that duplicate a list, modify it in one place, and expect the other copy to have those modifications, but there are <=11 such answers, so it should be fine. It's well worth the reduction in `LazyList` bugs and simplicity.

### PR DESCRIPTION
**Pull request by Lyxal**
Closes #833

### CODE REVIEW

File: vyxal/LazyList.py [hunk 1]
```diff
@@ -165,9 +165,7 @@ def __next__(self):
         return item
 
     def __setitem__(self, position, value):
```
Comment (cgccuser) [inline vyxal/LazyList.py:167]: We can just get rid of this method tbh, no need to implement it and raise a `NotImplementedError`

File: vyxal/LazyList.py [hunk 2]
```diff
@@ -165,9 +165,7 @@ def __next__(self):
...
-        if position >= len(self.generated):
-            self.__getitem__(position)
-        self.generated[position] = value
+        raise NotImplementedError("LazyList does not support assignment")
 
     def __delitem__(self, index):
```
Comment (cgccuser) [inline vyxal/LazyList.py:170]: This method can also be removed.

**Review by cgccuser — CHANGES_REQUESTED** [review]
This is great, just a couple methods to delete.

**Comment by cgccuser** [comment]
Should this be marked as a draft while that anything else that relies on mutability stuff gets modified?

**Comment by Seggan** [comment]
+1

File: vyxal/LazyList.py (hunk 2 above)
Comment (Lyxal) [reply vyxal/LazyList.py:170]: My idea was that having an error would help make it clear if there were any errors trying to assign to a lazy lists.

File: vyxal/LazyList.py (hunk 1 above)
Comment (Lyxal) [reply vyxal/LazyList.py:167]: To be removed later when we're sure we got every instance of mutability.

For now, it's a failsafe.

**Comment by cgccuser** [comment]
@Lyxal If the method doesn't exist, it's going to error anyway.

**Comment by Lyxal** [comment]
But at least with a custom error it's clearer what's going wrong

**Review by cgccuser — APPROVED** [review]

**Review by Seggan — APPROVED** [review]

**Pull request merged**